File: backend/CleanArch/WebApi/Dtos/ArchObject.cs
using Domain;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace WebApi.Dtos
{
    public class CreateArchObjectRequest : BaseLogDomain
    {
        public string? address { get; set; }
        public string? name { get; set; }
        public string? identifier { get; set; }
        public int? district_id { get; set; }
        public string? description { get; set; }
        public int[] tags { get; set; }
        public double? xcoordinate { get; set; }
        public double? ycoordinate { get; set; }
        public int? tunduk_district_id { get; set; }
        public int? tunduk_addres_unit_id { get; set; }
        public int? tunduk_street_id { get; set; }
        public string? tunduk_bulding_num { get; set; }
        public string? tunduk_flat_num { get; set; }
        public string? tunduk_uch_num { get; set; }
        public bool? is_manual { get; set; }
    }

    public class UpdateArchObjectRequest
    {
        public int id { get; set; }
        public string? address { get; set; }
        public string? name { get; set; }
        public string? identifier { get; set; }
        public int? district_id { get; set; }
        public string? description { get; set; }
        public int[] tags { get; set; }
        public double? xcoordinate { get; set; }
        public double? ycoordinate { get; set; }
        public int? tunduk_district_id { get; set; }
        public int? tunduk_address_unit_id { get; set; }
        public int? tunduk_street_id { get; set; }
        public string? tunduk_building_num { get; set; }
        public string? tunduk_flat_num { get; set; }
        public string? tunduk_uch_num { get; set; }
        public bool? is_manual { get; set; }

    }
}

File: frontend/src/constants/ArchObject.ts
export type ArchObject = {
  id: number;
  address: string;
  name: string;
  identifier: string;
  district_id: number;
  tunduk_district_id?: number;
  tunduk_address_unit_id?: number;
  tunduk_street_id?: number;
  tunduk_street_input_value?: string;
  tunduk_building_id?: number;
  tunduk_uch_num?: string;
  tunduk_flat_num?: string;
  tunduk_building_num?: string;
  open?: boolean;
  xcoordinate: number;
  ycoordinate: number;
  description: string;
  name_kg: string;
  tags: number[];
  district_name?: string;
  is_manual?: boolean;
};


export interface ArchObjectValues extends ArchObject {
  geometry: any[];
  addressInfo: AddresInfo[];
  point: any[];
  DarekSearchList: [];
  errordistrict_id: string;
  errortunduk_district_id?: string;
  errortunduk_address_unit_id?: string;
  errortunduk_street_id?: string;
  errortunduk_building_id?: string;
  errordescription: string;
  erroraddress: string;
  count?: number;
  legalRecords?:[]
  legalActs?:[]
}
export type AddresInfo = {
  address: string;
  propcode?: string;
}